app/src/shared/functions.ts

```diff
@@ -4,3 +4,13 @@ import { By } from '@angular/platform-browser';
 export const queryElement = (targetFixture: ComponentFixture<any>, targetClass: string) => {
   return targetFixture.debugElement.query(By.css(targetClass));
 };
+
+export const readFileAsString = (file: File): Promise<string> => {
+  const reader = new FileReader();
+  return new Promise((resolve) => {
+    reader.readAsText(file);
+    reader.onload = () => {
```

Comment:
In general, it's better to register event handlers through the `addEventListener` method. So for this you would do something like `reader.addEventListener('load', () => {`.

Reply:
Updated 👍